### DISCUSSION TITLE
UUID Description field

### DISCUSSION THREAD

**Discussion by stavrt** [emericg/toolBLEx#11] [Ideas]
It would be great to have a lookup table of UUIDs to a custom string description. When you connect and read all the services and characteristics, the description field shows what exactly each service  and description does.

**Comment by emericg** [comment]
It would be cool, but are you aware of such a list being available out there?

I can find services and characteristics names, but the app already shows that.
https://bitbucket.org/bluetooth-SIG/public/src/main/assigned_numbers/uuids/service_uuids.yaml
https://bitbucket.org/bluetooth-SIG/public/src/main/assigned_numbers/uuids/characteristic_uuids.yaml

Now string description for "known" services/characteristics is maybe possible to create but would be a huge amount of work, and "custom" services/characteristics description, by definition pretty unique to each device model, an even crazier amount of work.

**Reply by koenvervloesem** [reply]
I maintain a listing based on a couple of sources, published as a Python package: https://github.com/koenvervloesem/bluetooth-numbers

Edit: sorry, forgot this is not a Python project :-) But you can use https://github.com/NordicSemiconductor/bluetooth-numbers-database which my project is based on.

**Reply by emericg** [reply]
Hi, yes I know about your project actually :)
I probably have almost the same data you have already (in here https://github.com/emericg/toolBLEx/tree/master/assets/vendors), most of it directly from bluetooth.com (if I documented that correctly) and the oui are from https://maclookup.app/downloads/csv-database and heavily tweaked for speedy parsing.

But I can see you have a bit more than I do, so I'll have a look at it!
And thank you for the databases from Nordic, I don't think I was aware of them. And indeed they have json files which could be integrated as is (even if a good old csv would probably be half the size and faster to load, but these are not very big anyway)

**Reply by stavrt** [reply]
The list I was aware of. I was referring to custom services/characteristics. It could be a local table of string matching and the user would need to customize it, even if it was in JSON or YAML, the ability to override the UUID with another string.
